examples/step-71/step-71.cc

```diff
@@ -0,0 +1,937 @@
+      std::vector<Tensor<1, dim>>        exact_gradients(q_points.size());
+      exact_solution.gradient_list(q_points, exact_gradients);
+
+
+      // Positively, because we now only deal with one cell adjacent to the
+      // face (as we are on the boundary), the computation of the penalty
+      // factor $\gamma$ is substantially simpler:
+      const unsigned int p = fe.degree;
+      const double       gamma =
+        (1.0 * p * (p + 1) /
+         cell->extent_in_direction(
+           GeometryInfo<dim>::unit_normal_direction[face_no]));
+
+      // The third piece is the assembly of terms. This is now slightly more
+      // involved since these contains both terms for the matrix and for
+      // the right hand side. The latter requires us to evaluate the
+      // boundary conditions $j(\mathbf x)$, which in the current
+      // case (where we know the exact solution) we compute from
+      // $j(\mathbf x) = \frac{\partial u(\mathbf x)}{\partial {\mathbf n}}$:
+      for (unsigned int qpoint = 0; qpoint < q_points.size(); ++qpoint)
+        {
+          const auto &n = normals[qpoint];
+
+          for (unsigned int i = 0; i < n_dofs; ++i)
+            {
+              const double av_hessian_i_dot_n_dot_n =
+                (fe_interface_values.average_hessian(i, qpoint) * n * n);
+              const double jump_grad_i_dot_n =
+                (fe_interface_values.jump_gradient(i, qpoint) * n);
+
+              for (unsigned int j = 0; j < n_dofs; ++j)
+                {
+                  const double av_hessian_j_dot_n_dot_n =
+                    (fe_interface_values.average_hessian(j, qpoint) * n * n);
+                  const double jump_grad_j_dot_n =
+                    (fe_interface_values.jump_gradient(j, qpoint) * n);
+
+                  copy_data_face.cell_matrix(i, j) +=
+                    (-av_hessian_i_dot_n_dot_n  // - {grad^2 v n n}
+                       * jump_grad_j_dot_n      //   [grad u n]
+                                                //
+                     - av_hessian_j_dot_n_dot_n // - {grad^2 u n n}
+                         * jump_grad_i_dot_n    //   [grad v n]
+                                                //
+                     + gamma                    //  gamma
+                         * jump_grad_i_dot_n    // [grad v n]
+                         * jump_grad_j_dot_n    // [grad u n]
+                     ) *
+                    JxW[qpoint]; // dx
+                }
+
+              copy_data.cell_rhs(i) +=
+                (-av_hessian_i_dot_n_dot_n *       // - {grad^2 v n n }
+                   (exact_gradients[qpoint] * n)   //   (grad u_exact . n)
+                 +                                 // +
+                 gamma                             //  gamma
+                   * jump_grad_i_dot_n             // [grad v n]
+                   * (exact_gradients[qpoint] * n) // (grad u_exact . n)
+                 ) *
+                JxW[qpoint]; // dx
```

Comment:
Where does the first term come from?

Reply:
Hm, good question. @tjhei: Did we simply forget it in the description in the introduction, where we define the right hand side vector? I think so.